src/utils/asyncForEach.ts

```diff
@@ -0,0 +1,5 @@
+export async function asyncForEach<T>(array: T[], callback: (item: T, index: number, allItems: T[]) => void) {
```

Comment:
I don't like this asyncForEach.
You are using the async/await, you should avoid the callbacks